pkg/kubectl/cmd/clusterinfo_dump.go

```diff
@@ -0,0 +1,204 @@
+/*
+Copyright 2016 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmd
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"path"
+
+	"github.com/spf13/cobra"
+
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/client/unversioned"
+	"k8s.io/kubernetes/pkg/kubectl"
+	cmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
+)
+
+// NewCmdCreateSecret groups subcommands to create various types of secrets
+func NewCmdClusterInfoDump(f *cmdutil.Factory, cmdOut io.Writer) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "dump",
+		Short: "Dump lots of relevant info for debugging and diagnosis.",
+		Long:  dumpLong,
+		Run: func(cmd *cobra.Command, args []string) {
+			cmdutil.CheckErr(dumpClusterInfo(f, cmd, args, cmdOut))
+		},
+	}
+	cmd.Flags().String("output-directory", "", "Where to output the files.  If empty or '-' uses stdout, otherwise creates a directory hierarchy in that directory")
+	cmd.Flags().StringSlice("namespaces", []string{}, "A comma separated list of namespaces to dump.")
+	cmd.Flags().Bool("all-namespaces", false, "If true, dump all namespaces.  If true, --namespaces is ignored.")
+	return cmd
+}
+
+const (
+	dumpLong = `
+Dumps cluster info out suitable for debugging and diagnosing cluster problems.  By default, dumps everything to
+stdout. You can optionally specify a directory with --output-directory.  If you specify a directory, kubernetes will
+build a set of files in that directory.  By default only dumps things in the 'kube-system' namespace, but you can
+switch to a different namespace with the --namespaces flag, or specify --all-namespaces to dump all namespaces.
+`
+
+	dumpExample = `  # Dump current cluster state to stdout
+  kubectl cluster-info dump
+  
+  # Dump current cluster state to /tmp
+  kubectl cluster-info dump --output-directory=/tmp
+  
+  # Dump all namespaces to stdout
+  kubectl cluster-info dump --all-namespaces
+  
+  # Dump a set of namespaces to /tmp
+  kubectl cluster-info dump --namespaces default,kube-system --output-directory=/tmp`
```

Comment:
We should print a message showing that the cluster info is dumped (somewhere) successfully.